common/djangoapps/mailchimp_pipeline/signals/handlers.py

```diff
@@ -36,41 +36,44 @@ def update_mailchimp(email, data):
 @receiver(post_save, sender=EmailPreference)
 def sync_email_preference_with_mailchimp(sender, instance, **kwargs):
     email_preferences = instance
+    opt_in = ''
 
     if email_preferences.opt_in == 'yes':
         opt_in = 'TRUE'
     elif email_preferences.opt_in == 'no':
         opt_in = 'FALSE'
-    else:
-        opt_in = ''
+
     user_json = {
         "merge_fields": {
             "OPTIN": opt_in
         }
     }
 
-    update_mailchimp(instance.user.email, user_json)
+    update_mailchimp(email_preferences.user.email, user_json)
 
 
 @receiver(post_save, sender=UserProfile)
 def sync_user_profile_with_mailchimp(sender, instance, **kwargs):
     profile = instance
-    user_json = {
-        "merge_fields": {
-            "LANG": profile.language if profile.language else "",
-            "COUNTRY": profile.country.name.format() if profile.country else "",
-            "CITY": profile.city if profile.city else "",
+
+    if profile.language or profile.country or profile.city:
+        user_json = {
+            "merge_fields": {
+                "LANG": profile.language if profile.language else "",
+                "COUNTRY": profile.country.name.format() if profile.country else "",
+                "CITY": profile.city if profile.city else "",
+            }
         }
-    }
 
-    update_mailchimp(instance.user.email, user_json)
+        update_mailchimp(profile.user.email, user_json)
 
 
 @receiver(post_save, sender=UserExtendedProfile)
 def sync_extended_profile_with_mailchimp(sender, instance, **kwargs):
     extended_profile = instance
     org_label, org_type, work_area = get_org_data_for_mandrill(
         extended_profile.organization)
+
     user_json = {
         "merge_fields": {
             "ORG": org_label,
```

Comment:
This is not one of the lines I made any change to. This will cause inconsistency with the rest of the file/module, so I avoided doing that and still strongly advice against it, since that is not in the scope of the ticket, too.

Reply:
This makes sense that you didn't change these lines, But whole module is already inconsistent with quotes, So we can make it consistent by making changes where we do some refactoring.